tensorflow/python/data/benchmarks/benchmark_base.py

```diff
@@ -56,15 +56,34 @@ def run_benchmark(self,
       This is the median time (with respect to `iters`) it takes for the dataset
       to go through `num_elements` elements, divided by `num_elements.`
     """
-    options = dataset_ops.Options()
+
+    # The options that have been applied to the dataset are preserved so that
+    # they are not overwritten while benchmarking.
+    options = dataset.options()
     options.experimental_optimization.apply_default_optimizations = (
         apply_default_optimizations)
     dataset = dataset.with_options(options)
+
     # NOTE: We use `dataset.skip()` to perform the iterations in C++, avoiding
-    # the overhead of multiple `session.run()` calls. Note that this relies on
-    # the underlying implementation of `skip`: if it is optimized in the future,
-    # we will have to change this code.
+    # the overhead of multiple `session.run()` calls (in graph mode). Note that
+    # this relies on the underlying implementation of `skip`: if it is optimized
+    # in the future, we will have to change this code.
     dataset = dataset.skip(num_elements - 1)
+
+    if context.executing_eagerly():
+      deltas = []
+      for _ in range(iters):
+        if warmup:
+          iterator = iter(dataset)
+          next(iterator)
+
+        iterator = iter(dataset)
+        start = time.time()
+        next(iterator)
+        end = time.time()
+        deltas.append(end - start)
+      return np.median(deltas) / float(num_elements)
+
     iterator = dataset_ops.make_initializable_iterator(dataset)
     next_element = iterator.get_next()
     next_element = nest.flatten(next_element)[0]
```

Comment:
@jsimsa  Is the nest.flatten() transformation necessary? I have not added this in the eager case.

Reply:
@jsimsa setting/unsetting the `TF2_BEHAVIOR` env var didn't make a difference. I had to change the .bazelrc file and modify the config from:
```
build --config=v2
test --config=v2
```
to:
```
build --config=v1
test --config=v1
```
Then it picked up the config saying:
```
Found applicable config definition build:v1 in file /home/vikoth18/tensorflow/.bazelrc: --define=tf_api_version=1 --action_env=TF2_BEHAVIOR=0
```
If there is an alternative or a better approach please let me know. I am currently running the benchmarks and will post the outputs soon.